src/Hooks/useDappBookmarksList/useDappBookmarksList.ts

```diff
@@ -0,0 +1,51 @@
+import { useMemo } from "react"
+import { DiscoveryDApp } from "~Constants"
+import { useVeBetterDaoDapps } from "~Hooks"
+import {
+    selectCustomDapps,
+    selectFavoriteRefs,
+    selectFavoritesDapps,
+    selectFeaturedDapps,
+    useAppSelector,
+} from "~Storage/Redux"
+import uniqBy from "lodash/uniqBy"
+import { resolveDAppsFromReferences } from "~Utils/DAppUtils/DAppBookmarkResolver"
+
+export const useDappBookmarksList = (): DiscoveryDApp[] => {
+    const favoriteRefs = useAppSelector(selectFavoriteRefs)
+    const featuredDapps = useAppSelector(selectFeaturedDapps)
+    const customDapps = useAppSelector(selectCustomDapps)
+    const oldFavorites = useAppSelector(selectFavoritesDapps)
+    const { data: vbdDapps = [] } = useVeBetterDaoDapps()
+
+    return useMemo(() => {
+        // Fallback for pre-migration users
+        if (!favoriteRefs || favoriteRefs.length === 0) {
+            const dapps: DiscoveryDApp[] = [...oldFavorites, ...customDapps]
+            return uniqBy(dapps, value => value.href)
+        }
```

Comment:
I believe this case won't happen, migration occurs at startup

Reply:
Good catch.